pkg/sql/catalog/descpb/structured.proto

```diff
@@ -1108,15 +1108,20 @@ message TableDescriptor {
                               (gogoproto.customname) = "JobID", deprecated = true];
   }
 
+  // Before 22.1:
   // The schema elements that have been dropped and whose underlying
   // data needs to be gc-ed. These schema elements have already transitioned
   // through the drop state machine when they were in the above mutations
   // list, and can be safely deleted. The names for these schema elements
   // can be reused. This list is separate because mutations can
   // lie in this list for a long time (gc deadline) and should not block
   // the execution of other schema changes on the table.
+  //
+  // Since 22.1 this is field is deprecated and no longer maintained.
+  // The index GC job still removes mutations it finds in this list.
   repeated GCDescriptorMutation gc_mutations = 33 [(gogoproto.nullable) = false,
-                                                  (gogoproto.customname) = "GCMutations"];
+                                                   (gogoproto.customname) = "GCMutations",
+                                                   deprecated = true];
```

Comment:
I believe you could get away with outright removing this field (well, replacing it with a `reserved 33` to prevent reuse of this tag number in this proto) and then remove `updateDescriptorGCMutations` outright. Sure, any descriptors might still have a populated `GCMutations` around, but:
- it's a handful of bytes at worst,
- if it's never read, it doesn't matter what's in it,
- plus it will disappear the next time that the table descriptor is updated. 

This shouldn't cause any problems in a mixed-version cluster either. Worst case the GC job gets picked up by a node with the old version, which then edits this slice when it's done. It kind of expects the GC'd index to be in the slice, but doesn't complain if it's not.

Reply:
Happy to go either way here.  What would SQL prefer?